src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -425,32 +428,20 @@ public void DeleteAttachmentBeforeRevert(DocumentsOperationContext context, Lazy
             }
         }
 
-        public void RevisionAttachments(DocumentsOperationContext context, Slice lowerId, Slice changeVector)
+        public void RevisionAttachments(DocumentsOperationContext context, BlittableJsonReaderObject document, Slice lowerId, Slice changeVector)
         {
-            using (GetAttachmentPrefix(context, lowerId.Content.Ptr, lowerId.Size, AttachmentType.Document, Slices.Empty, out Slice prefixSlice))
+            var attachmentsFromDocumentMetadata = GetAttachmentsFromDocumentMetadata(document)
+                .Select(attachment => JsonDeserializationClient.AttachmentName(attachment)).ToList();
+            foreach (var attachment in attachmentsFromDocumentMetadata)
             {
-                var table = context.Transaction.InnerTransaction.OpenTable(AttachmentsSchema, AttachmentsMetadataSlice);
-                var currentAttachments = new List<(LazyStringValue name, LazyStringValue contentType, Slice base64Hash)>();
-                foreach (var sr in table.SeekByPrimaryKeyPrefix(prefixSlice, Slices.Empty, 0))
```

Comment:
We are now ignoring what we have on storage? What if a user just removed an entry from the metadat?